vit-servicing-station-lib/migrations/2020-05-22-112032_initial/down.sql

```diff
@@ -3,5 +3,5 @@ DROP TABLE IF EXISTS proposals;
 DROP TABLE IF EXISTS funds;
 DROP TABLE IF EXISTS voteplans;
 DROP TABLE IF EXISTS api_tokens;
+DROP TABLE IF EXISTS challenges;
```

Comment:
I don't think modifying old migrations is a proper way to do it in diesel.
Migrations should be applied (or deapplied) sequentially and recorded in a meta-table, so any newly introduced change should be created as a new migration applying the delta to the effective schema after the previous migration..

Reply:
I think in this case its ok since we are using a different version from fund to fund.

Reply:
> I don't think modifying old migrations is a proper way to do it in diesel.

Agree

Reply:
This is actually problematic. Writing a proper `down.sql` script for the new migration requires `ALTER TABLE proposals DROP COLUMN challenge_id`, but SQLite don't support `DROP COLUMN`. You can copy the columns that should be left after a rollback, but that looks a bit odd.

Reply:
> You can copy the columns that should be left after a rollback, but that looks a bit odd.

It's a bit odd indeed, but the workaround [looks simple enough](https://stackoverflow.com/questions/5938048/delete-column-from-sqlite-table).

Reply:
> I think in this case its ok since we are using a different version from fund to fund.

True that, also our database is created once and never reused in different deployments.

Should we scrap the whole migration system for simplicity, then? Making changes to timestamped migrations looks like misuse to me. Maybe we should remove the timestamp cruft from the single migration directory's name and name it something prominently like `migrations/init`.

Reply:
We use the migration because the library helps in loading the scripts for testing purposes and for the CLI tool to initiate an empty DB. Mostly so we do not have to do it by hand or using some other script for that. 
We can check if we can rename it removing the timestamp and moving to another directory as you comment. It should be ok.